### ISSUE TITLE
If this library is imported into an app that already uses prometheus and initializes the client, then the second client init will fail because of re-registering the enum

### ISSUE THREAD

**Issue by wcn00** [apache/pulsar-client-go#332]
#### Expected behavior

Tell us what should happen

#### Actual behavior

Tell us what happens instead

#### Steps to reproduce

How can we reproduce the issue

#### System configuration
**Pulsar version**: x.y

**Comment by merlimat** [comment]
@wcn00 Do you think that is something related to how we're importing or initializing prometheus stuff? 

Is this something related with https://github.com/golang/protobuf/issues/178#issuecomment-373136277 ?

According to https://github.com/golang/protobuf/issues/268#issuecomment-593773447, there should have been some improvement in protobuf registration in 1.20.0

**Comment by wcn00** [comment]
@merlimat  Yes I do think it has something to do with those issues.   I would migrate to the gogo version of protobufs, but I don't control either of these packages.  For the time being I'll build the app against a modified version of prometheous and leave it to upstream teams do change their code.
thnx
wcn
